discovery-provider/plugins/notifications/src/processNotifications/mappers/milestone.ts

```diff
@@ -139,16 +139,68 @@ export class Milestone extends BaseNotification<MilestoneRow> {
   }
 
   getResourcesForEmail(): ResourceIds {
+    let tracks = new Set<number>()
+    let playlists = new Set<number>()
+    if (this.type === MilestoneType.TRACK_REPOST_COUNT || this.type === MilestoneType.TRACK_SAVE_COUNT) {
+      const data = this.notification.data as TrackMilestoneNotification
+      tracks.add(data.track_id)
+    }
+    else if (this.type === MilestoneType.PLAYLIST_REPOST_COUNT || this.type === MilestoneType.PLAYLIST_SAVE_COUNT) {
+      const data = this.notification.data as PlaylistMilestoneNotification
+      playlists.add(data.playlist_id)
+    }
     return {
       users: new Set([this.receiverUserId]),
+      tracks,
+      playlists
     }
   }
 
   formatEmailProps(resources: Resources) {
-    const receiverUserId = resources.users[this.receiverUserId]
+    const receiverUser = resources.users[this.receiverUserId]
+    let achievement
+    let entity
+    if (this.type === MilestoneType.FOLLOWER_COUNT) {
+      achievement = 'follow'
```

Comment:
tiny nit - do we wanna hard code these achievements (follow, repost, etc) as enum constants somewhere? or do they not already exist?

Reply:
we should at some point, will leave as is for now